tests/TestCase/View/Helper/PaginatorHelperTest.php

```diff
@@ -3495,4 +3495,32 @@ protected function setPagingParams($params, bool $merge = true)
         }
         $this->View->setRequest($this->View->getRequest()->withAttribute('paging', $params));
     }
+
+    /**
+     * Test that generated URLs work without sort defined within the request
+     *
+     * @return void
+     */
+    public function testPaginationWithPassedParams()
+    {
+        $request = new ServerRequest([
+            'url' => '/articles/index/whatever/3',
+            'params' => [
+                'plugin' => null, 'controller' => 'articles', 'action' => 'index', 'pass' => ['whatever', '3']
```

Comment:
```suggestion
                'plugin' => null, 'controller' => 'Articles', 'action' => 'index', 'pass' => ['whatever', '3']
```

Reply:
If controller param is capitalized the url result changes to: Articles/index...